From 7220000a73392069473b815428d1140b9bb3600a Mon Sep 17 00:00:00 2001
From: Peter Ujfalusi <peter.ujfalusi@ti.com>
Date: Mon, 6 Oct 2014 17:24:40 +0530
Subject: [PATCH 1271/1587] clk: ti: clk-7xx: Configure ATL1 and ATL2 clocks

Set the parent of the atl_gfclk_mux and configure the rate
for atl1 and atl2.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@ti.com>
Signed-off-by: Lokesh Vutla <lokeshvutla@ti.com>
Acked-by: Tero Kristo <t-kristo@ti.com>
[zou: Original patch taken from
ti-sdk-am335x-evm-08.00.00.00-Linux-x86-Install.bin]
Signed-off-by: Cao Zou <cao.zou@windriver.com>
---
 drivers/clk/ti/clk-7xx.c |   19 ++++++++++++++++++-
 1 files changed, 18 insertions(+), 1 deletions(-)

diff --git a/drivers/clk/ti/clk-7xx.c b/drivers/clk/ti/clk-7xx.c
index f37043e..5b7a474 100644
--- a/drivers/clk/ti/clk-7xx.c
+++ b/drivers/clk/ti/clk-7xx.c
@@ -21,7 +21,7 @@
 #define DRA7_DPLL_DSP_DEFFREQ				600000000
 #define DRA7_DPLL_DSP_GFCLK_NOMFREQ			600000000
 #define DRA7_DPLL_EVE_GCLK_NOMFREQ			400000000
-
+#define DRA7_ATL_DEFFREQ				5644800
 
 static struct ti_dt_clk dra7xx_clks[] = {
 	DT_CLK(NULL, "atl_clkin0_ck", "atl_clkin0_ck"),
@@ -311,6 +311,7 @@ int __init dra7xx_dt_clk_init(void)
 {
 	int rc;
 	struct clk *abe_dpll_mux, *sys_clkin2, *dpll_ck, *dss_deshdcp_ck;
+	struct clk *atl_fck, *atl_parent;
 	struct clk *ipu1_gfclk, *ipu1_gfclk_parent;
 	struct clk *dsp_dpll, *dsp_m2_dpll, *dsp_m3x2_dpll;
 
@@ -343,6 +344,22 @@ int __init dra7xx_dt_clk_init(void)
 	if (rc)
 		pr_err("%s: failed to enable DESHDCP clock\n", __func__);
 
+	atl_fck = clk_get_sys(NULL, "atl_gfclk_mux");
+	atl_parent = clk_get_sys(NULL, "dpll_abe_m2_ck");
+	rc = clk_set_parent(atl_fck, atl_parent);
+	if (rc)
+		pr_err("%s: failed to reparent atl_gfclk_mux\n", __func__);
+
+	atl_fck = clk_get_sys(NULL, "atl_clkin2_ck");
+	rc = clk_set_rate(atl_fck, DRA7_ATL_DEFFREQ);
+	if (rc)
+		pr_err("%s: failed to set atl_clkin2_ck\n", __func__);
+
+	atl_fck = clk_get_sys(NULL, "atl_clkin1_ck");
+	rc = clk_set_rate(atl_fck, DRA7_ATL_DEFFREQ);
+	if (rc)
+		pr_err("%s: failed to set atl_clkin1_ck\n", __func__);
+
 	ipu1_gfclk = clk_get_sys(NULL, "ipu1_gfclk_mux");
 	ipu1_gfclk_parent = clk_get_sys(NULL, "dpll_core_h22x2_ck");
 	rc = clk_set_parent(ipu1_gfclk, ipu1_gfclk_parent);
-- 
1.7.5.4

